inference-engine/src/gna_plugin/gna_graph_patterns.hpp

```diff
@@ -12,6 +12,45 @@
 
 namespace GNAPluginNS {
 
+inline bool IsReshapeFrom4dTo3d(InferenceEngine::CNNLayerPtr layer) {
+    if (!LayerInfo(layer).isNonFunctional()) {
+        return false;
+    }
+
+    auto input_dims = layer->insData[0].lock()->getDims();
+    auto output_dims = layer->outData[0]->getDims();
+    if (input_dims.size() != 4 || output_dims.size() != 3 ||
+        input_dims[2] != 1) {
```

Comment:
why it can't be 1 here? I would suggest adding comments in such a cases

Reply:
this function checks if it's the Reshape inserted after Convolution to reshape it from NCHW to NCW, if H !=1 it can't be just skipped, so it's not a Reshape which we expect to find
added additional comments